src/cogent3/app/io_new.py

```diff
@@ -150,28 +132,33 @@ def open_data_store(
     -------
     ReadOnlyDirectoryDataStore or ReadOnlyZippedDataStore
     """
+    if not isinstance(suffix, (str, type(None))):
+        raise ValueError(f"suffix {type(suffix)} not one of string or None")
+
+    kwargs = dict(limit=limit, mode=mode, suffix=suffix)
     base_path = Path(base_path)
     base_path = base_path.expanduser().absolute()
-    if base_path.suffix in (".tinydb", ".sqlitedb"):
-        suffix = "json"
-
-    if suffix is None:
-        raise ValueError("suffix required")
-
-    if not base_path.exists():
-        raise ValueError(f"'{base_path}' does not exist")
+    if base_path.suffix == ".tinydb":
+        kwargs["suffix"] = "json"
+        kwargs.pop("mode")
 
-    if type(suffix) != str:
-        raise ValueError(f"{suffix} is not a string")
-
-    if zipfile.is_zipfile(base_path):
+    if base_path.suffix == ".sqlitedb":
```

Comment:
If we use `base.path.suffix`, why we have an argument `suffix` to this function?

Reply:
suffix is for the directory data store class, which identifies members  as those with the nominated ending